File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numbers = set(nums)
        result = 0
        
        for num in numbers:
            if (num - 1) not in numbers:
                current = 0
                while (num + current) in numbers:
                    current += 1
                if current > result:
                    result = current

        return result

File: Data Structures & Algorithms/sum-of-two-integers/submission-2.java
class Solution {
    public int getSum(int a, int b) {
        if (b == 0) {
            return a;
        }
        else if (a == 0) {
            return b;
        }
        int carry =  a & b;
        int result = a ^ b;
        while(carry != 0) {
            int shiftedcarry = carry << 1;
            carry = result & shiftedcarry;
            result ^= shiftedcarry;
        }
        return result;
    }
}
